src/main/java/calculator/StringAddCalculator.java

```diff
@@ -0,0 +1,55 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    public static int splitAndSum(String text) {
+
+        if (isValidationCheckNullOrEmpty(text)) {
+            return 0;
+        }
+
+        if (isNegativeNumbers(text)) {
+            throw new RuntimeException("음수 값은 입력할 수 없습니다.");
+        }
+
+        Integer result = customDelimiterAddCalculator(text);
+        if (result != null) {
+            return result;
+        }
+
+        return standardStringAddCalculator(text);
+    }
+
+    private static boolean isValidationCheckNullOrEmpty(String text) {
+        return text == null || text.isEmpty();
+    }
+
+    private static int standardStringAddCalculator(String text) {
+        String[] tokens = text.split(",|:");
+
+        return getSum(tokens);
+    }
+
+    private static boolean isNegativeNumbers(String text) {
+        return text.contains("-");
+    }
+
+    private static Integer customDelimiterAddCalculator(String text) {
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
```

Comment:
현재 메서드 호출할때마다 Pattern.compile()을 호출하고 있네요 :)

Pattern.compile 내부 구현을 보면
```
// Pattern
public static Pattern compile(String regex) {
  return new Pattern(regex, 0);
}
```

매번 메서드 호출시 성능이 비싼 Pattern 인스턴스를 생성하고 있기에
Pattern 인스턴스는 한번 쓰고 버려지기 때문에 성능적으로 좋지 않다고 생각해요 :)
자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있는 것 같아요!

```
If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time
```

불필요한 객체 생성을 피하도록 개선해주시면 좋을 것 같아요 😄

Reply:
인스턴스 비용이나 생성에 대해서는 생각해본 적이 없었네요
피드백 감사드립니다! 👍